q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+    """Check that the user-defined value for the
+    rule "validation" is correctly formatted.
+
+    validation must be a dict
+        Apply the rule of validating variables values: replace by the blank
+        (or the missing) value every sample entry that for the current variable has
+        in one or multiple other variable(s) a null or missing value for that sample.
+                    validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "validation".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    # accepted structure for the "validation" rule
+    possible_validations = {
+        'force_to_blank_if': {'is null'}
+    }
+    # check the first level is a dict
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
+            # get the possible conditions
+            possible_validations_2 = possible_validations['force_to_blank_if']
+            # for each of the condition/replacements for the accepted keys
+            for rule_2, rule_value_2 in rule_value['force_to_blank_if'].items():
+                # check that the content of replacements are lists
+                if isinstance(rule_value_2, list):
+                    # get the accepted conditions from the current conditions
+                    if rule_2 in possible_validations_2:
+                        return True
+    return False
+
+
+def check_normalization(rule_value):
+    """Check that the user-defined value for the
+    rule "normalization" is correctly formatted.
+
+    normalization must be a dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "normalization".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, dict):
+        possible_normalizations = {'maximum', 'minimum', 'gated_value'}
+        rule_value_set = set(rule_value)
+        if not len(rule_value_set ^ (rule_value_set & possible_normalizations)):
+            return True
+    return False
+
+
+def check_format(rule_value):
```

Comment:
same here... I must be missing something? I'm not seeing it as part of what is already merged either.

Reply:
this is the function name that is "found" by the above `rule_method = globals()[rule_method_key]`.
This have been changed anyway as above review, and it is now more obvious in:
```
    rule_type_map = {
        'expected': (check_expected, ExpectedError, 'lookups', ),
        'ontology': (check_ontology, OntologyError, 'lookups'),
        'remap': (check_remap, RemapError, 'edits'),
        'validation': (check_validation, ValidationError, 'edits'),
        'normalization': (check_normalization, NormalizationError, 'edits'),
        'blank': (check_blank_missing, BlankError, 'allowed'),
        'missing': (check_blank_missing, MissingError, 'allowed'),
        'format': (check_format, FormatError, 'allowed')
    }

    if rule not in rule_type_map:
        raise RuleError(rule, variable, rule_value)

    rule_method, rule_error, rule_type = rule_type_map[rule]
    if rule_method(rule_value):
        rules[rule_type][rule] = rule_value
    else:
        raise rule_error(variable, rule_value)

```